examples/js/archive.js

```diff
@@ -352,13 +382,111 @@ downloadJSON.addEventListener('click', () => {
     }
 });
 
+// save JSON to localStorage
+saveMapBtn.addEventListener('click', () => {
+  const jsonImages = map.imgGroup.generateExportJson(true);
+  const d = new Date();
+  const datetime = d.getHours() + ":" + d.getMinutes() + ' ' + d.getDate() + "/" + (d.getMonth() + 1) + "/" + d.getFullYear();
+
+  if (jsonImages.images.length) {
+    const modifiedJsonImages = {};
+    const tempCollection = [];
+
+    // restructure jsonImages
+    modifiedJsonImages.avg_cm_per_pixel = jsonImages.avg_cm_per_pixel;
+    jsonImages.images.map((image) => {
+      tempCollection.push({
+        id: image.id,
+        src: image.src,
+        tooltipText: image.tooltipText,
+        image_file_name: image.image_file_name,
+        nodes: image.nodes,
+        cm_per_pixel: image.cm_per_pixel,
+      });
+    });
+    modifiedJsonImages.collection = tempCollection;
+    let savedMaps = []
+    const newMap = {
+      map: modifiedJsonImages,
+      timeSaved: datetime,
+      amountOfImages: modifiedJsonImages.collection.length
+    };
+
+    // this is to check if are old maps saved, if true, update with new maps to be saved 
+    if (JSON.parse(localStorage.getItem('savedMaps'))) {
+      const oldMaps = JSON.parse(localStorage.getItem('savedMaps'));
+      savedMaps = [...oldMaps, newMap]
+      localStorage.setItem('savedMaps', JSON.stringify(savedMaps));
+    } else {
+      savedMaps = [newMap]
+      localStorage.setItem('savedMaps', JSON.stringify(savedMaps));
+    }
+    console.log('savedMap', savedMaps)
+    bootstrap.Modal.getInstance(savedMapsModal).hide();
+    alert("Saved!")
+   
+  }
+});
+
 // share map modal
 const shareModal = document.getElementById('shareModal')
 const modality =  new bootstrap.Modal(shareModal)
 shareMapBtn.addEventListener('click', () => {
   bootstrap.Modal.getInstance(shareModal).show()
 });
 
+// history map modal
```

Comment:
This is some complex but relatively self contained code which would be great to move into a module in a later PR!

Reply:
My thoughts exactly @jywarren